src/hotspot/cpu/riscv/templateTable_riscv.cpp

```diff
@@ -2008,7 +2023,15 @@ void TemplateTable::fast_binaryswitch() {
     // else [i = h]
     // Convert array[h].match to native byte-ordering before compare
     __ shadd(temp, h, array, temp, 3);
-    __ ld(temp, Address(temp, 0));
+    if (AvoidUnalignedAccesses) {
+      // array is BytesPerInt (aka 4) aligned
+      __ lwu(t1, Address(temp,4));
+      __ slli(t1, t1, 32);
+      __ lwu(temp, Address(temp, 0));
+      __ add(temp, temp, t1);
+    } else {
+      __ ld(temp, Address(temp, 0));
+    }
```

Comment:
This if-else could be further simplified into a single "__ lwu(temp, Address(temp, 0));" since only the low 32 bits of 'temp' register is used by following 'revb_w_w'.